sgid_client/SgidClient.py

```diff
@@ -0,0 +1,214 @@
+import secrets
+from typing import TypedDict
+from urllib.parse import urlparse, urlencode
+import requests
+from .validation import validate_access_token, validate_id_token
+from .IdTokenVerifier import IdTokenVerifier
+from .decrypt_data import decrypt_data
+from .error import Errors, get_network_error_message, get_www_authenticate_error_message
+from .util import convert_to_pkcs8
+
+API_VERSION = 2
+
+
+class AuthorizationUrlReturn(TypedDict):
+    url: str
+    nonce: str | None
+
+
+class CallbackReturn(TypedDict):
+    sub: str
+    access_token: str
+
+
+class UserInfoReturn(TypedDict):
+    sub: str
+    data: dict
+
+
+class SgidClient:
+    """Class which allows you to interact with the sgID API."""
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        private_key: str,
+        redirect_uri: str | None,
+        hostname: str = "https://api.id.gov.sg",
+    ):
+        """Initialises an SgidClient instance.
+
+        Args:
+            client_id (str): Client ID provided during client registration.
+
+            client_secret (str): Client secret provided during client registration
+
+            private_key (str): Client private key provided during client registration
```

Comment:
nit: 
```suggestion
            client_secret (str): Client secret provided during client registration.

            private_key (str): Client private key provided during client registration.
```

some comments have full stops and some don't - let's standardize to have full stops for all? :D

Reply:
b3a03db